### PR TITLE
Make `_atom_type.element_symbol` the enumeration default on some data items

### LINKED ISSUE #401: Why are most `templ_enum` default value listings keyed on `_atom_type.symbol`?
Many of the `_enumeration_default.index` save frames in `templ_enum` are keyed on `_atom_type.symbol`.
   - `atomic_number`
   - `electron_count`
   - `ion_to_element`
   - `atomic_mass`
   - `radius_bond`
   - `length_neutron`
   - `dispersion_real_cu`, `dispersion_imag_cu`
   - `dispersion_real_mo`, `dispersion_imag_mo`
   - `cromer_mann_a1` -- `cromer_mann_c`
   - `hi_ang_fox_c0` -- `hi_ang_fox_c3`

The last two, have truely different values for each `_atom_type.symbol`, as does `electron_count`, and `ion_to_element`.

The remainder just repeat the same information for each ion of an element; that is, it's elemental information, not ionic (even though the bond radius should be different? What's the history there?) (ignoring `length_neutron` here)

Why can't these other enumerations be indexed on `_atom_type.element_symbol`, which is populated from `_atom_type.symbol` by `ion_to_element`?

(One could also conceive of `_atom_type.charge_number`, which is taken directly from the `_atom_type.symbol`, and then electron_count could be calculated from atomic_number and charge_number.)

### PR DESCRIPTION
**Pull request by rowlesmr**
Will close #401 

Dataitems done in this commit:
   - ~~`atomic_number`~~
   - ~~`atomic_mass`~~
   - `radius_bond`
      - Shouldn't these be ion-specific, or isn't that the intent here? cf charge vs oxidation in #378 
   - `length_neutron`
      - This will be the topic of another PR
   - ~~`dispersion_real_cu`, `dispersion_imag_cu`~~
   - ~~`dispersion_real_mo`, `dispersion_imag_mo`~~

Dataitems not being touched:
   - `electron_count`
   - `ion_to_element`
   - `cromer_mann_a1` -- `cromer_mann_c`
   - `hi_ang_fox_c0` -- `hi_ang_fox_c3`

### CODE REVIEW

File: templ_enum.cif [hunk 1]
```diff
@@ -870,49 +870,32 @@ save_atomic_number
      loop_
```
Comment (vaitkus) [inline templ_enum.cif:870]: Shouldn't the atomic mass also depend on the mass number? Maybe the changes to this frame/item should be delayed until we can actually specify the multivalue key?

Comment (rowlesmr) [reply templ_enum.cif:870]: Yes, it should.

File: cif_core.dic [hunk 2]
```diff
@@ -27439,4 +27439,9 @@ save_
         method.
 
         Added _atom_type.mass_number to record specific isotope identities.
+
+        Changed enumeration of _atom_type.atomic_number, _atom_type.atomic_mass,
+        _atom_type_scat.dispersion_real_Cu, _atom_type_scat.dispersion_imag_Cu,
+        _atom_type_scat.dispersion_real_Mo, _atom_type_scat.dispersion_imag_Mo
+        to be defaulted on _atom_type.element_symbol.
```
Comment (vaitkus) [inline cif_core.dic:27446]:
```suggestion
        Changed enumeration of _atom_type.atomic_number,
        _atom_type_scat.dispersion_real_Cu, _atom_type_scat.dispersion_imag_Cu,
        _atom_type_scat.dispersion_real_Mo, _atom_type_scat.dispersion_imag_Mo
        to be defaulted on _atom_type.element_symbol.
```

File: cif_core.dic [hunk 3]
```diff
@@ -23799,7 +23799,7 @@ save_
 save_atom_type.atomic_mass
 
     _definition.id                '_atom_type.atomic_mass'
-    _definition.update            2012-11-20
+    _definition.update            2023-06-01
```
Comment (vaitkus) [inline cif_core.dic:23802]:
```suggestion
    _definition.update            2012-11-20
```

File: templ_enum.cif [hunk 4]
```diff
@@ -2416,4 +2311,8 @@ save_
 
        Updated the values for V2+, V3+ and V5+ in the 'length_neutron' save
        frame to match those of V.
+
+       Altered key of atomic_number, atomic_mass, dispersion_real_Cu,
```
Comment (vaitkus) [inline templ_enum.cif:2315]:
```suggestion
       Altered key of atomic_number, dispersion_real_Cu,
```

**Review by vaitkus — APPROVED** [review]
Other than the outlined minor issues the PR looks great.

**Review by jamesrhester — APPROVED** [review]

**Pull request merged**